tests/test_layabout.py

```diff
@@ -1 +1,327 @@
-import layabout  # noqa: F401
+import os
+from unittest.mock import MagicMock, call
+
+import pytest
+
+from layabout import (
+    Layabout,
+    MissingSlackToken,
+    FailedConnection,
+)
+
+# The likelihood of this ever being a valid Slack token is probably slim.
+TOKEN = 'xoxb-13376661337-DeAdb33Fd15EA53fACef33D1'
+
+
+@pytest.fixture
+def slack_client():
+    def api_call(method):
+        return dict(
+            channels=[{'id': 'D3ADB33F1', 'name': 'general'}],
+            groups=[{'id': '1800IDGAF', 'name': 'super_secret_club'}],
+        )
+
+    slack_client = MagicMock(api_call=api_call)
+    return slack_client
+
+
+def test_layabout_can_register_handlers():
+    """
+    Test that layabout can register Slack API event handlers normally and as a
+    decorator, both with and without keyword arguments.
+    """
+    layabout = Layabout(name='test')
+    kwargs = dict(spam='🍳')
+
+    def handle_hello1(slack, event):
+        pass
+
+    @layabout.handle('hello')
+    def handle_hello2(slack, event):
+        pass
+
+    def handle_hello3(slack, event, spam):
+        pass
+
+    @layabout.handle('hello', kwargs=kwargs)
+    def handle_hello4(slack, event, spam):
+        pass
+
+    layabout.handle(type='hello')(fn=handle_hello1)
+    layabout.handle(type='hello', kwargs=kwargs)(fn=handle_hello3)
```

Comment:
There aren't any assertions. The test could conceivably fail if the inner workings of `Layabout` were screwed up enough, but it's unlikely.

We actually had a separate conversation about this test in particular where you thought I was being overly specific. The original test _did_ include assertions, but became troublesome. I was registering handlers and then reaching into the private `self._handlers` object to verify they were present in the right way. I was also verifying that the functions that had been registered were exactly the same as their definitions, but that broke down because of the decorators wrapping and changing their `id`s.

Reply:
Oh, that makes sense.  Then at this point, this test is pretty much just an executed example of different ways to register event handlers.  In that case, I might put in an assertion that there are 4 handlers registered. To me that would communicate a bit better what the test is about, though it doesn't really change what it's testing. Either way is fine though.